### ISSUE TITLE
Exit on startup: bad-txns-vout-empty, with 2 patches applied

### ISSUE THREAD

**Issue by Stadicus** [lightningnetwork/lnd#1670]
### Background

LND exits on startup, due to dust channels. I get the known "bad-txns-vout-empty" error, but even after applying the diff patch (see https://github.com/lightningnetwork/lnd/issues/975#issuecomment-377613760) a startup is not possible. LND exits after detecting a channel in "state 2".

My Raspberry Pi node was offline for about a week, so there should not be any unconfirmed channels.

Noteworthy log excerpts:

#### official lnd 0.4.2-beta release
```
2018-08-02 19:43:56.135 [INF] LTND: Version 0.4.2-beta commit=7cf5ebe2650b6798182e10be198c7ffc1f1d6e19
2018-08-02 19:43:56.136 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2018-08-02 19:43:56.138 [INF] CHDB: Checking for schema update: latest_version=0, db_version=0

2018-08-02 19:44:29.000 [INF] LNWL: Finished rescan for 757 addresses (synced to block 00000000000000000004eb69ca8cb1b29af48cb40c42081cdc6042ce64b4fb2b, height 534880)
2018-08-02 19:44:29.003 [DBG] LNWL: Could not resend transaction 7a71a1a15569f6df1135188f3ac6f9087925cfba5a2b299e41a9d96ced926517: -26: 16: bad-txns-vout-empty
2018-08-02 19:44:30.868 [INF] HSWC: Payment circuits loaded: num_pending=1124, num_open=12
2018-08-02 19:44:32.504 [INF] HSWC: Trimming open circuits for chan_id=533688:570:1, start_htlc_id=0
2018-08-02 19:44:32.504 [INF] HSWC: Trimming open circuits for chan_id=515601:732:0, start_htlc_id=0
...
2018-08-02 19:44:32.507 [INF] HSWC: Trimming open circuits for chan_id=533687:2878:0, start_htlc_id=0
2018-08-02 19:44:33.934 [DBG] FNDG: channel (879667c62e909b08256d0aaf98cb4add244cc0af896a76d72308a8ce93f67266) with opening state 2 found
2018-08-02 19:44:33.934 [DBG] FNDG: Will announce channel 587281046371172352 after ChannelPoint(6672f693cea80823d7766a89afc04c24dd4acb98af0a6d25089b902ec6679687:0) has gottenconfirmations
...
2018-08-02 19:44:34.231 [INF] HSWC: Starting HTLC Switch
2018-08-02 19:44:34.231 [INF] FNDG: Announcing ChannelPoint(6672f693cea80823d7766a89afc04c24dd4acb98af0a6d25089b902ec6679687:0), short_chan_id=(*lnwire.ShortChannelID)(0x12640)(534129:2118:0)

2018-08-02 19:44:34.674 [DBG] HSWC: Closed completed FAIL circuit for 7360089a233fb0772088fba6b31a132f9ca7d5f3be074a238cf12ada3cb025bb: (518505:1502:1, 927) <-> (531357:1147:7)
2018-08-02 19:44:34.674 [DBG] HSWC: Closed completed FAIL circuit for 5029c91608e013bb96aad252f1137d1f36923e441e5a0369103bf4348bb311fb: (518505:1502:1, 929) <-> (531357:1147:8)
2018-08-02 19:44:34.674 [DBG] HSWC: Closed completed FAIL circuit for deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef: (518505:1502:1, 932) <-> (531357:1147:9)
2018-08-02 19:44:34.675 [DBG] HSWC: Closed completed FAIL circuit for deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef: (518505:1502:1, 934) <-> (531357:1147:10)
...
2018-08-02 19:44:44.965 [INF] UTXN: UTXO Nursery is now fully synced
2018-08-02 19:44:46.020 [INF] CNCT: Creating ChannelArbitrators for 30 active channels
2018-08-02 19:44:46.020 [DBG] CNCT: New ChainEventSubscription(id=0) for ChannelPoint(85883f6543458e864d5c226c672735bc3f48ac6dc5c1cd6bff21bd3d4f972a98:1)
2018-08-02 19:44:46.020 [INF] NTFN: New block epoch subscription
...
2018-08-02 19:44:46.201 [DBG] CNCT: Starting chain watcher for ChannelPoint(69b98e70c551e5fec6663bb30f32958de9e1159b67a031ad7bdab840d35db96e:0)
2018-08-02 19:44:46.202 [INF] NTFN: New spend subscription: utxo=69b98e70c551e5fec6663bb30f32958de9e1159b67a031ad7bdab840d35db96e:0
2018-08-02 19:44:46.259 [DBG] CNCT: Starting chain watcher for ChannelPoint(85b4a04efa1c8ce2ae11f5fda2223a882d767f040e3a2573337383bd27838995:1)
2018-08-02 19:44:46.260 [INF] NTFN: New spend subscription: utxo=85b4a04efa1c8ce2ae11f5fda2223a882d767f040e3a2573337383bd27838995:1
2018-08-02 19:44:46.259 [INF] CNCT: Close observer for ChannelPoint(69b98e70c551e5fec6663bb30f32958de9e1159b67a031ad7bdab840d35db96e:0) active
2018-08-02 19:44:46.359 [DBG] CNCT: Starting chain watcher for ChannelPoint(e47934b305be49d296f205aa99b3fcafc51d880c70bcdfdbd28c0e7c342e887f:1)
2018-08-02 19:44:46.360 [INF] NTFN: New spend subscription: utxo=e47934b305be49d296f205aa99b3fcafc51d880c70bcdfdbd28c0e7c342e887f:1
2018-08-02 19:44:46.359 [INF] CNCT: Close observer for ChannelPoint(85b4a04efa1c8ce2ae11f5fda2223a882d767f040e3a2573337383bd27838995:1) active
...
2018-08-02 19:44:50.212 [INF] HSWC: Removing channel link with ChannelID(7e2e340b14f49334efd56bda118fcd71719528cf357fdb273f0b4de970cb7cd2)
2018-08-02 19:44:50.213 [ERR] CNCT: unable to mark link inactive: channel link not found
2018-08-02 19:44:50.229 [INF] CNCT: Broadcasting force close transaction, ChannelPoint(d27ccb70e94d0b3f27db7f35cf28957171cd8f11da6bd5ef3493f4140b342e7e:0): (*wire.MsgTx)(0x12260)({
 Version: (int32) 2,
 TxIn: ([]*wire.TxIn) (len=1 cap=1) {
  (*wire.TxIn)(0x12262f80)({
   PreviousOutPoint: (wire.OutPoint) d27ccb70e94d0b3f27db7f35cf28957171cd8f11da6bd5ef3493f4140b342e7e:0,
   SignatureScript: ([]uint8) {
   },
   Witness: (wire.TxWitness) (len=4 cap=4) {
    ([]uint8) <nil>,
    ([]uint8) (len=72 cap=144) {
     00000000  30 45 02 21 00 d9 d0 0b  bc 22 9b 66 67 0f 87 aa  |0E.!.....".fg...|
     00000010  a1 25 f5 65 3e 58 5c 5a  0c 10 16 5c 2d 8d b5 58  |.%.e>X\Z...\-..X|
     00000020  33 90 39 d2 5e 02 20 6f  3d 5d 23 41 6d 89 3c 7c  |3.9.^. o=]#Am.<||
     00000030  26 83 9d a4 23 4d 45 7b  3e f5 97 5e 06 ee 6a 0e  |&...#ME{>..^..j.|
     00000040  ff 2e 9f 87 a5 a6 aa 01                           |........|
    },
    ([]uint8) (len=72 cap=144) {
     00000000  30 45 02 21 00 89 41 b5  99 e5 1c 57 ff 73 7d fc  |0E.!..A....W.s}.|
     00000010  94 0b e9 fc cf 72 f6 1c  7c 9a 18 8a 97 e1 38 e7  |.....r..|.....8.|
     00000020  11 b1 3f 47 12 02 20 69  16 d2 b0 18 6c 8f ea 6a  |..?G.. i....l..j|
     00000030  74 d6 32 e9 f2 08 34 e0  c2 12 9a a0 96 80 24 50  |t.2...4.......$P|
     00000040  7a 11 a8 49 52 91 72 01                           |z..IR.r.|
    },
    ([]uint8) (len=71 cap=500) {
     00000000  52 21 03 58 53 99 ac 22  ec 55 02 df 41 3e 41 f6  |R!.XS..".U..A>A.|
     00000010  ec f4 da 74 03 3b 88 96  fd ec e8 5b 88 6e fd 03  |...t.;.....[.n..|
     00000020  22 b8 2c 21 03 5a af 1c  e5 16 1c b9 ea 56 1d 78  |".,!.Z.......V.x|
     00000030  cf 85 ab 44 c2 d1 46 93  97 2c 4c 52 6a a7 59 ce  |...D..F..,LRj.Y.|
     00000040  3e 3f 1f 4c c1 52 ae                              |>?.L.R.|
    }
   },
   Sequence: (uint32) 2161036595
  })
 },
 TxOut: ([]*wire.TxOut) {
 },
 LockTime: (uint32) 550869010
})

2018-08-02 19:44:50.234 [ERR] CNCT: ChannelArbitrator(d27ccb70e94d0b3f27db7f35cf28957171cd8f11da6bd5ef3493f4140b342e7e:0): unable to broadcast close tx: -26: 16: bad-txns-voumpty
2018-08-02 19:44:50.234 [ERR] CNCT: unable to advance state: -26: 16: bad-txns-vout-empty
2018-08-02 19:44:50.235 [ERR] SRVR: unable to start server: -26: 16: bad-txns-vout-empty
```
At this point, LND exits

#### LND compiled from master (1.08.2018) with diff patch
https://github.com/lightningnetwork/lnd/issues/975#issuecomment-377613760
![grafik](https://user-images.githubusercontent.com/19550140/43603705-0c2c4bb2-9694-11e8-9406-61420b9397bb.png)

```
2018-08-02 19:50:00.157 [INF] LTND: Version 0.4.2-beta commit=f6ea91af713546174bf04a79302116aa455d89c8
2018-08-02 19:50:00.158 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2018-08-02 19:50:00.160 [INF] CHDB: Checking for schema update: latest_version=3, db_version=0
2018-08-02 19:50:00.160 [INF] CHDB: Performing database schema migration
2018-08-02 19:50:00.161 [INF] CHDB: Applying migration #1
2018-08-02 19:50:00.161 [INF] CHDB: Populating new node update index bucket
2018-08-02 19:50:01.641 [INF] CHDB: Populating new edge update index bucket
2018-08-02 19:50:27.783 [INF] CHDB: Migration to node and edge update indexes complete!
2018-08-02 19:50:27.783 [INF] CHDB: Applying migration #2
2018-08-02 19:50:27.784 [INF] CHDB: Migrating invoice database to new time series format
2018-08-02 19:50:27.795 [INF] CHDB: Migration to invoice time series index complete!
2018-08-02 19:50:27.795 [INF] CHDB: Applying migration #3
2018-08-02 19:50:27.796 [INF] CHDB: Migrating invoice database to new outgoing payment format
...
2018-08-02 19:50:54.453 [DBG] LNWL: Could not resend transaction 7a71a1a15569f6df1135188f3ac6f9087925cfba5a2b299e41a9d96ced926517: -26: 16: bad-txns-vout-empty
2018-08-02 19:50:55.161 [INF] HSWC: Payment circuits loaded: num_pending=1124, num_open=12
2018-08-02 19:50:55.511 [INF] HSWC: Trimming open circuits for chan_id=533688:570:1, start_htlc_id=0
2018-08-02 19:50:55.512 [INF] HSWC: Trimming open circuits for chan_id=515601:732:0, start_htlc_id=0
... 
2018-08-02 19:50:55.514 [INF] HSWC: Trimming open circuits for chan_id=533687:2878:0, start_htlc_id=0
2018-08-02 19:50:56.938 [DBG] FNDG: channel (879667c62e909b08256d0aaf98cb4add244cc0af896a76d72308a8ce93f67266) with opening state 2 found
2018-08-02 19:50:56.939 [DBG] FNDG: Will announce channel 587281046371172352 after ChannelPoint(6672f693cea80823d7766a89afc04c24dd4acb98af0a6d25089b902ec6679687:0) has gotten 6 confirmations
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x23e8f8]

goroutine 138 [running]:
github.com/lightningnetwork/lnd/chainntnfs.(*TxConfNotifier).Register(0x0, 0x13190720, 0x0, 0x0)
        /home/admin/go/src/github.com/lightningnetwork/lnd/chainntnfs/txconfnotifier.go:121 +0x20
github.com/lightningnetwork/lnd/chainntnfs/bitcoindnotify.(*BitcoindNotifier).RegisterConfirmationsNtfn(0x1275a050, 0x127fa000, 0x13ac8e00, 0x22, 0x1f4, 0x6, 0x8266c, 0x128262c0, 0x1, 0x0)
        /home/admin/go/src/github.com/lightningnetwork/lnd/chainntnfs/bitcoindnotify/bitcoind.go:799 +0x15c
main.(*fundingManager).annAfterSixConfs(0x12ef80d0, 0x23be9180, 0x1262bb30, 0x13ab4570, 0x0)
        /home/admin/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:2104 +0x298
main.(*fundingManager).Start.func4(0x12ef80d0, 0x1262bb30, 0x12a6d0b8, 0x23be9180)
        /home/admin/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:682 +0x58
created by main.(*fundingManager).Start
        /home/admin/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:679 +0x6f0
```

#### LND compiled from master (2.08.2018) with additional patch (if c.cfg.ForceCloseChan == nil)
https://github.com/lightningnetwork/lnd/issues/1306#issuecomment-398122172
![grafik](https://user-images.githubusercontent.com/19550140/43603526-7f20aaba-9693-11e8-8130-5f37b03325b0.png)

Now I get almost the same error message as before:
```
2018-08-02 20:29:31.093 [INF] HSWC: Trimming open circuits for chan_id=533687:2878:0, start_htlc_id=0
2018-08-02 20:29:32.535 [DBG] FNDG: channel (879667c62e909b08256d0aaf98cb4add244cc0af896a76d72308a8ce93f67266) with opening state 2 found
2018-08-02 20:29:32.535 [DBG] FNDG: Will announce channel 587281046371172352 after ChannelPoint(6672f693cea80823d7766a89afc04c24dd4acb98af0a6d25089b902ec6679687:0) has gotten 6 confirmations
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x23e8f8]

goroutine 102 [running]:
github.com/lightningnetwork/lnd/chainntnfs.(*TxConfNotifier).Register(0x0, 0x1206a1e0, 0x0, 0x0)
        /home/admin/go/src/github.com/lightningnetwork/lnd/chainntnfs/txconfnotifier.go:121 +0x20
github.com/lightningnetwork/lnd/chainntnfs/bitcoindnotify.(*BitcoindNotifier).RegisterConfirmationsNtfn(0x132bc0a0, 0x12018000, 0x132a7200, 0x22, 0x1f4, 0x6, 0x8266c, 0x15fd0, 0x2, 0x11f2a1c0)
        /home/admin/go/src/github.com/lightningnetwork/lnd/chainntnfs/bitcoindnotify/bitcoind.go:799 +0x15c
main.(*fundingManager).annAfterSixConfs(0x129080d0, 0x11f3f400, 0x12a55990, 0x12a07c80, 0x398c6c)
        /home/admin/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:2104 +0x298
main.(*fundingManager).Start.func4(0x129080d0, 0x12a55990, 0x11eb0850, 0x11f3f400)
        /home/admin/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:682 +0x58
created by main.(*fundingManager).Start
        /home/admin/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:679 +0x6f0
```

### Your environment
* lnd version 0.4.2-beta commit=f6ea91af713546174bf04a79302116aa455d89c8
* Linux RaspiBolt 4.14.52-v7+ #1123 SMP Wed Jun 27 17:35:49 BST 2018 armv7l GNU/Linux
* Bitcoin Core Daemon version v0.16.2

### Steps to reproduce
Start LND as describe above:
* official release
* with 1st patch
* with additional 2 patch

### Expected behaviour
LND starts and lets me close channels that are not in a viable state anymore.

### Actual behaviour
LND exits directly on start, so no user action is possible

**Comment by wpaulino** [comment]
The latest panic that you're hitting will be fixed by #1669.

**Comment by Roasbeef** [comment]
#1669 has been merged, closing.
